oeps/oep-0022-bp-django-caches.rst

```diff
@@ -0,0 +1,229 @@
+decorator(s) to use.
+
+Sample Before::
+
+    def get_and_cache_my_business_value(x):
+        cached_value = cache.get(key, _CACHE_MISS)
+        if cached_value is _CACHE_MISS:
+            new_value = calculate_my_business_value()
+            key = get_my_business_key(x)
+            cache.set(key, new_value)
+            return new_value
+        return cached_value
+
+Sample After::
+
+    @sample_cache_decorator(['x'])
+    def get_and_cache_my_business_value(x):
+        # calculate my_business_value
+        ...
+        return my_business_value
+
+See `Implementations`_ for more details.
+
+Generating Cache Keys
+~~~~~~~~~~~~~~~~~~~~~
+
+Use a utility function that will provide a Memcached safe key that will avoid
+key conflicts in the global space in Memcached. This functionality may or may
+not come as part of a caching decorator.
+
+See `Implementations`_ for more details.
+
+Implementations
+~~~~~~~~~~~~~~~
+
+This section will be updated as a shared library is selected and evolved.
+
+**In edx-platform:**
+
+* There is no utility for handling the `Common Caching Defect and Fix`_.
+
+* There is an `edx-platform RequestCache`_, but no Tiered-Cache.
+
+* There are no utilities for forcing cache misses.
+
+* There are various cache decorators. These could be assessed when considering
+  creating a shared library of caching utilities.
+
+* There are common utilities for generating cache keys. These should be be
+  assessed when considering creating a shared library of caching utilities.
+
+**In ecommerce:**
+
+* The `RequestCache and TieredCache`_ both handle the `Common Caching Defect
+  and Fix`_. Although it is simple to refactor standard Django caching code
+  to use these classes, the interface to these classes is non-standard and
+  the simple refactor doesn't move toward using cache decorators.
+
+* There is both a `RequestCache and TieredCache`_.
+
+* The TieredCache provides functionality for forcing cache misses.
+
+* There is no cache decorator utility.
+
+* There are common utilities for generating cache keys. These should be be
+  assessed when considering creating a shared library of caching utilities.
+
+**Quickcache library**
+
+Note: The `quickcache library`_ has not been fully assessed. It is provided as a
+good potential candidate for a shared library.
+
+* The `Common Caching Defect and Fix`_ is handled by using a decorator. Any calls
+  to the Django cache outside of the decorator would be susceptible to the bug.
+
+* Provides a TieredCache, but no RequestCache. The library would need to be
+  enhanced to include the RequestCache, or to extend to use one.
+
+* There is functionality for skipping the cache. However, this functionality still
+  needs to be assessed.
+
+* The library is based on a slick @quickcache decorator. The decorator is
+  opinionated about its cache keys, tied to the code it is wrapping. This needs
+  to be assessed to see if it is a good fit.
```

Comment:
Assessed or extended, if it doesn't meet our needs.  As mentioned above, a decorator could theoretically be extended by taking in a function that computes the cache-key.

Reply:
You can read my update soon and let me know what you think.